scripts/train.py

```diff
@@ -0,0 +1,206 @@
+    callbacks.append(lr_scheduler)
+
+    return callbacks
+
+
+def create_generators(args):
+    # create image data generator objects
+    train_image_data_generator = keras.preprocessing.image.ImageDataGenerator(
+        horizontal_flip=True,
+    )
+    val_image_data_generator = keras.preprocessing.image.ImageDataGenerator()
+
+    if args.dataset_type == 'coco':
+        # import here to prevent unnecessary dependency on cocoapi
+        from keras_retinanet.preprocessing.coco import CocoGenerator
+
+        train_generator = CocoGenerator(
+            args.coco_path,
+            'train2017',
+            train_image_data_generator,
+            batch_size=args.batch_size
+        )
+
+        validation_generator = CocoGenerator(
+            args.coco_path,
+            'val2017',
+            val_image_data_generator,
+            batch_size=args.batch_size
+        )
+    elif args.dataset_type == 'pascal':
+        train_generator = PascalVocGenerator(
+            args.pascal_path,
+            'trainval',
+            train_image_data_generator,
+            batch_size=args.batch_size
+        )
+
+        validation_generator = PascalVocGenerator(
+            args.pascal_path,
+            'test',
+            val_image_data_generator,
+            batch_size=args.batch_size
+        )
+    elif args.dataset_type == 'csv':
+        train_generator = CSVGenerator(
+            args.csv_path,
+            args.classes_path,
+            image_datrain_image_data_generator,
+            batch_size=args.batch_size
+        )
+
+        if args.val_path:
+            validation_generator = CSVGenerator(
+                args.val_path,
+                args.classes_path,
+                val_image_data_generator,
+                batch_size=args.batch_size
+            )
+        else:
+            validation_generator = None
+    else:
+        raise ValueError('Invalid data type received: {}'.format(dataset_type))
+
+    return train_generator, validation_generator
+
+
+def parse_args():
+    parser     = argparse.ArgumentParser(description='Simple training script for training a RetinaNet network.')
+    subparsers = parser.add_subparsers(help='Arguments for specific dataset types.', dest='dataset_type')
+    subparsers.required = True
+
+    coco_parser = subparsers.add_parser('coco')
+    coco_parser.add_argument('coco_path', help='Path to dataset directory (ie. /tmp/COCO).')
+
+    pascal_parser = subparsers.add_parser('pascal')
+    pascal_parser.add_argument('pascal_path', help='Path to dataset directory (ie. /tmp/VOCdevkit).')
+
+    csv_parser = subparsers.add_parser('csv')
+    csv_parser.add_argument('csv_path', help='Path to CSV file for training.')
+    csv_parser.add_argument('classes_path', help='Path to a CSV file containing class label mapping.')
+    csv_parser.add_argument('--val_path', help='Path to CSV file for validation (optional).')
```

Comment:
Should be `--val-path`?

Also, `csv_path` isn't very descriptive since `classes_path` also designates a CSV file. How abbout `annotations` and `classes`? That together with a good description should be clear enough.

Reply:
:+1: